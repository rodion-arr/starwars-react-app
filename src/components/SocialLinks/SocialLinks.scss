.social-links {
  &__text {
    text-align: center;
    text-transform: uppercase;
  }

  .social-links-list {
    list-style: none;
    padding: 0;
    display: flex;
    align-items: center;
    justify-content: center;

    &__item {
      width: 25px;
      height: 25px;
      margin: 0 10px;

      & .icon-wrap,
      & .social-links-list__icon {
        display: block;
        width: 100%;
        height: 100%;
      }

      &--linkedin {
        .social-links-list__icon {
          background: no-repeat center/ cover url("../../images/footer/linkedin-logo.svg");
        }
      }

      &--github {
        .social-links-list__icon {
          background: no-repeat center/ cover url("../../images/footer/github-logo.svg");
        }
      }
    }
  }
}
