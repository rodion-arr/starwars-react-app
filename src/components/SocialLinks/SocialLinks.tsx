import './SocialLinks.scss';
import React from "react";

export const SocialLinks = (): JSX.Element => {
  return (
    <div className="social-links">
      <p className="social-links__text">Follow:</p>
      <ul className="social-links-list">
        <li className="social-links-list__item">
          <a className="social-links-list__item--linkedin icon-wrap" href="https://www.linkedin.com/in/rodionabdurakhimov/" target="_blank" rel="noopener noreferrer" title="Rodion Abdurakhimov | LinkedIn">
            <i className="social-links-list__icon" />
          </a>
        </li>
        <li className="social-links-list__item">
          <a className="social-links-list__item--github icon-wrap" href="https://github.com/rodion-arr" target="_blank" rel="noopener noreferrer" title="Rodion Abdurakhimov | GitHub">
            <i className="social-links-list__icon" />
          </a>
        </li>
      </ul>
    </div>
  );
};
