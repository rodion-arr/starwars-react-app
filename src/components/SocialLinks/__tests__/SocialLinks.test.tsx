import {render} from "@testing-library/react";
import {SocialLinks} from "../SocialLinks";

describe('SocialLinks', () => {
  it('should render correctly', function () {
    const {container} = render(<SocialLinks />);

    expect(container).toMatchSnapshot();
  });
});
