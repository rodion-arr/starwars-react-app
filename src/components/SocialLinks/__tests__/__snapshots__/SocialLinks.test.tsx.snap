// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks should render correctly 1`] = `
<div>
  <div
    class="social-links"
  >
    <p
      class="social-links__text"
    >
      Follow:
    </p>
    <ul
      class="social-links-list"
    >
      <li
        class="social-links-list__item"
      >
        <a
          class="social-links-list__item--linkedin icon-wrap"
          href="https://www.linkedin.com/in/rodionabdurakhimov/"
          rel="noopener noreferrer"
          target="_blank"
          title="Rodion Abdurakhimov | LinkedIn"
        >
          <i
            class="social-links-list__icon"
          />
        </a>
      </li>
      <li
        class="social-links-list__item"
      >
        <a
          class="social-links-list__item--github icon-wrap"
          href="https://github.com/rodion-arr"
          rel="noopener noreferrer"
          target="_blank"
          title="Rodion Abdurakhimov | GitHub"
        >
          <i
            class="social-links-list__icon"
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
